File: src/main/java/cz/grunwjir/csasdemo/gmaps/client/GMapsClientProperties.java
package cz.grunwjir.csasdemo.gmaps.client;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Configuration properties class for Google Maps API client.
 *
 * @author Jiri Grunwald
 */
@Getter
@Setter
@ConfigurationProperties(prefix = "gmaps")
public class GMapsClientProperties {

    private String apiKey;
}

File: src/main/kotlin/cz/grunwjir/csasdemo/gmaps/client/GMapsClientProperties.kt
package cz.grunwjir.csasdemo.gmaps.client

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

/**
 * Configuration properties class for Google Maps API client.
 *
 * @author Jiri Grunwald
 */

@ConstructorBinding
@ConfigurationProperties(prefix = "gmaps")
data class GMapsClientProperties(val apiKey: String)